caikit/core/toolkit/errors/error_handler.py

```diff
@@ -291,19 +291,25 @@ def value_check(self, log_code, condition, *args, msg=""):
                 A boolean value that should describe if this check passes `True` or fails `False`.
                 Upon calling this function, this is typically provided as an expression, e.g.,
                 `0 < variable < 1`.
-            msg:  str
-                A string message describing the value check that failed.  If the empty string is
-                provided (default) then no additional information will be provided.  Note that
-                string interpolation can be lazily performed on `msg` using `{}` format syntax by
-                passing additional arguments.  This is the preferred method for performing string
-                interpolation on `msg` so that it is only done if an error condition is encountered.
             *args:
-                Additional `{}`-style string formatting arguments to be lazily interpolated into
-                the `msg` argument.
+                A variable set of arguments describing the value check that failed. If only
+                1 argument is provided then an empty msg string is assumed and no additional
+                information will be provided, otherwise the first argument will be treated as 'msg'
+                argument. Note that string interpolation can be lazily performed on `msg` using `{}`
+                format syntax by passing additional arguments.  This is the preferred method for
+                performing string interpolation on `msg` so that it is only done if an error
+                condition is encountered.
+
         """
         if not get_config().enable_error_checks:
             return
 
+        if len(args) == 1:
+            msg = "{}"
```

Comment:
I would thing this would look more like the following:

```py
if not condition:
    msg = (
        "" if not args else (
            args[0] if len(args) == 1 else args[0].format(args[1:])
        )
    )
```

This way, all the `if` conditions are only evaluated after the check on `condition`. Also, no extra formatting is done (even "empty" formatting is not free).

One advantage to the way you have it written is that `args[0]` could be something besides a string and it would be stringified by `msg.format`. Not sure that was intentional, though?

Reply:
Nope, wasn't my intention to stringify non-string params (do we need to?)

And yep incorporating all checks within the condition makes total sense!

Reply:
Had to tweak the above code to `format(*args[1:])` for unpacking the args

Reply:
Although this still fails if the user provides 2 args, neither of which are supposed to be strings.
```
bad_value = 1.1
self.error.value_check("<CCC>", 0.0 <= bad_value <= 1.0, bad_value, "<1.0")
```
results in an error since we try to format assuming `bad_value` as `msg`.

Reply:
We can additionally check if the type of arg is `str` before trying to `.format` with it?